fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/ImplementInJoinRule.java

```diff
@@ -140,48 +147,45 @@ public void onMatch(@Nonnull final CascadesRuleCall call) {
             final var providedOrdering = planPartition.getAttributeValue(OrderingProperty.ORDERING);
 
             for (final RequestedOrdering requestedOrdering : requestedOrderings) {
-                final ImmutableList<InSource> sources =
-                        getInSourcesForRequestedOrdering(explodeAliasToQuantifierMap, explodeAliases,
+                final var sourcesStream =
```

Comment:
The changes in this file are related to actually creating all possible orderings if the requested ordering is exhaustive. Orderings can be manipulated by pre-sorting the `InSource` of an `in-join` plan. By creating ascending and descending orders we create new possibilities if the result of this `in-join` partakes in a distinct set operation.